### ISSUE TITLE
Log parsing issues

### ISSUE THREAD

**Issue by tobiasdiez** [SublimeText/LaTeXTools#995]
LaTeXTools does not recognize a few "undefined control sequence" and "Missing character: There is no ... in font ... " errors and complains with
> (Log parsing issues. Disregard unless something else is wrong.)

in the build output. I use the basic builder with pdflatex + bibtex.

The log is attached (with txt-ending to be able to upload it to github). Let me know if you need further details.
[index.log.txt](https://github.com/SublimeText/LaTeXTools/files/691508/index.log.txt)

**Comment by diego898** [comment]
For some reason I am seeing this now as well, and am no longer getting warnings about missing references, etc.

**Comment by r-stein** [comment]
We slightly changed the parser code in the last patch #1099. I have no experience with this part of the code, but if it does more harm than good I can revert the changes.

**Comment by msiniscalchi** [comment]
@tobiasdiez I tried running your log file through `parseTeXlog` from the command line and everything was OK... do you still see the error?

@diego898 the output on pastebin looks good to me---there is no parsing error there.

So in either case I wonder what might be going on. Do you still see the problem after restarting ST?

**Comment by diego898** [comment]
@msiniscalchi Restarting ST (and computer) doesnt solve the problem. I searched the output of `parseTeXlog` as well but thought I was missing something.

**Comment by msiniscalchi** [comment]
@diego898 would it be possible for you to upload a test case (tex file)?

**Comment by diego898** [comment]
Hm this is strange - I can't seem to generate a MWE. I created a simple root file with a simple file import and cant replicate the log parsing warning.

**Comment by msiniscalchi** [comment]
@diego898 that's not surprising... your original document must hit some obscure corner of the parsing code. Can you upload the tex log file instead?

**Comment by diego898** [comment]
Here it is: https://pastebin.com/raw/1ZYx2n2s

Note: I have the following two lines in main file: 

```
\RequirePackage{silence}
\WarningFilter{caption}{Unsupported document class}
```

But I've tried commenting them out and I get the annoying caption warning, and still get the log parsing issue

**Comment by msiniscalchi** [comment]
This is weird. I thought it might be a python2 / python3 issue, but I ran your log through `parseTeXlog` using both (from the command line) and things worked fine. Unfortunately, without a tex test case, I can't really look into it :(

**Comment by diego898** [comment]
Hm. Do you know of a package/tool/method to replace all of the non-tex words in your document with words from lorem-ipsum? That way I can upload my "thesis" without any privacy concerns

**Comment by msiniscalchi** [comment]
Not really. The only other thing that may help is to copy the LaTeXTools build output from ST and put that on pastebin. This way I can try to figure out where the problem is in the log file you sent me.

**Comment by diego898** [comment]
This is my build output:

```
[Compiling /Users/diego/Dropbox/My Research/Publications/My Project/my_project.tex]

Basic Builder: running pdflatex...done.
running bibtex...done.
running pdflatex...done.
running pdflatex...done.
running pdflatex...done.

No errors. Warnings:

(Log parsing issues. Disregard unless something else is wrong.)

[Done!]
```

**Comment by msiniscalchi** [comment]
That's it??? Hmm...

Let's try something else: can you use the default builder instead of the basic builder?

**Comment by diego898** [comment]
When I select `Build with Latex - Basic Builder` I still get:

```
[Compiling /Users/diego/Dropbox/My Research/Publications/My Project/my_project.tex]

Basic Builder: running pdflatex...done.
running bibtex...done.
running pdflatex...done.
running pdflatex...done.
running pdflatex...done.

No errors. Warnings:

(Log parsing issues. Disregard unless something else is wrong.)

[Done!]
```

**Comment by msiniscalchi** [comment]
that's the basic builder. How about the default (aka traditional) builder?

**Comment by ig0774** [comment]
@diego898: You can get some more useful info for us if you edit the `parseTeXlog.py` file. To edit it run the **Browse Packages** command, open the LaTeXTools folder and it should be there. Near the top, there's a line that says `print_debug = False`. Change the `False` to `True` and save the file. You should then be able to run a build and a bunch of debugging information will be dumped to the Sublime console (not the build panel). If you can paste that here or on pastebin, that might help narrow down the problem.

**Comment by diego898** [comment]
Wow - I really cannot explain this. I continued working with my document and now I no longer see the log parsing issues warning!

Either way, I went ahead and set `debug` to `True` in `parseTeXlog` and here is the output:

https://pastebin.com/raw/6MjsLjWg

If it happens again, Ill post a new version of the log

**Comment by ig0774** [comment]
Hmmmm... well... two things could've happened: 1) turning on the debug statements actually enables a set of file-recognition heuristics that are useful for troubleshooting logs so we don't have to exactly emulate the install location of TeX and friends. In some rare cases, however, these heuristics cause some problems to disappear (they're not enabled more generally because they can yield false positives, resulting in the parser recognising errors as occurring in files that do not actually exist). 2) The most frequent cause of parser issues is due to trying to keep track of the opening and closing parentheses that indicate LaTeX opening or closing a file. Unfortunately, a lot of miscellaneous messages also contain parenthesis which can sometimes cause the parser to either prematurely think a file is closed or to miss the fact that the file was closed at all. The parser tries to be a bit conservative about this, but it sometimes results in parsing errors. Often small changes to documents can result in wildly different log messages, so sometimes these problems seem to go away on their own.

We probably need to build a check for parser errors to try to capture the log file and the corresponding debug information in a format that could easily be posted to GitHub to allow us to better capture and analyse (and hopefully eliminate) these issues.

**Comment by diego898** [comment]
Just to be complete, I have since turned off debug, but the error is still gone.

**Comment by diego898** [comment]
It came back! I then set `debug` to `True`, and it disappeared again. Setting it back to `False` makes it return now.
